contracts/zero-ex/contracts/src/features/MultiplexFeature.sol

```diff
@@ -0,0 +1,799 @@
+                ITransformERC20Feature.TransformERC20Args memory args;
+                args.inputToken = IERC20TokenV06(fillData.tokens[i]);
+                args.outputToken = IERC20TokenV06(fillData.tokens[i + 1]);
+                args.minOutputTokenAmount = 0;
+                args.taker = payable(_computeHopRecipient(fillData.calls, i));
+                if (nextTarget != address(0)) {
+                    // If `nextTarget` was set in the previous hop, then the input
+                    // token was already sent to the FlashWallet. Setting
+                    // `inputTokenAmount` to 0 indicates that no tokens need to
+                    // be pulled into the FlashWallet before executing the
+                    // transformations.
+                    args.inputTokenAmount = 0;
+                } else if (
+                    args.taker != msg.sender &&
+                    !args.inputToken.isTokenETH()
+                ) {
+                    address flashWallet = address(
+                        ITransformERC20Feature(address(this)).getTransformWallet()
+                    );
+                    // The input token has _not_ already been sent to the
+                    // FlashWallet. We also want PayTakerTransformer to
+                    // send the output token to some address other than
+                    // msg.sender, so we must transfer the input token
+                    // to the FlashWallet here.
+                    _transferERC20Tokens(
+                        args.inputToken,
+                        msg.sender,
+                        flashWallet,
+                        outputTokenAmount
+                    );
+                    args.inputTokenAmount = 0;
+                } else {
+                    // Otherwise, either:
+                    // (1) args.taker == msg.sender, in which case
+                    //     `_transformERC20` will pull the input token
+                    //     into the FlashWallet, or
+                    // (2) args.inputToken == ETH_TOKEN_ADDRESS, in which
+                    //     case ETH is attached to the call and no token
+                    //     transfer occurs.
+                    args.inputTokenAmount = outputTokenAmount;
+                }
+                uint256 ethValue;
+                (args.transformations, ethValue) = abi.decode(
+                    wrappedCall.data,
+                    (ITransformERC20Feature.Transformation[], uint256)
+                );
+                // Do not spend more than the remaining ETH.
+                ethValue = LibSafeMathV06.min256(ethValue, remainingEth);
+                // Call `_transformERC20`.
+                outputTokenAmount = ITransformERC20Feature(address(this))
+                    ._transformERC20{value: ethValue}(args);
+                // Decrement the remaining ETH.
+                remainingEth -= ethValue;
```

Comment:
Where do unspent protocol fees ETH go?

Reply:
back to the taker –– the actual taker, which may differ from `args.taker` here, so we'd set the FQT `refundReceiver` to the actual taker's address rather than `address(1)`

Reply:
hm I guess we wouldn't be able to do that if the taker address isn't passed to the API, but...
let's just not fill limit orders from within a multihop fill 🤝